### DISCUSSION TITLE
Won't it be possible to server stale content from proxy-cache plugin anyway?

### DISCUSSION THREAD

**Discussion by khaled4vokalz** [Kong/kong#12738] [Help]
The DOC (https://docs.konghq.com/hub/kong-inc/proxy-cache/#upstream-outages) states that the proxy-cache can't serve stale content currently

> Due to an implementation in Kong’s core request processing model

So, is there any plan on mitigating this? or is it possible to do it on our own? Or there's no way, as the core doesn't support it...

**Comment by StarlightIbuki** [comment]
What the document says is that when the upstream is out, and all the local caches are out of life, it cannot serve any content.
The following sentence suggests setting a larger `storage_ttl` to make the cache persist longer. Note that the cache lifetime is different from what we usually refer to as HTTP cache lifetime.

**Reply by khaled4vokalz** [reply]
@StarlightIbuki ,
I was more talking about features like this e.g. https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_background_update .... setting a large storage_ttl and large cache_control header doesn't always make sense right? It'd have been nice if we could handle contents that gets updated in short time spans.... Currently It's not possible from KONG to tackle highloads on upstreams on a stale cache scenario, but nginx provides that out of the box....

**Reply by StarlightIbuki** [reply]
I got your point. However, the only possible solution I can think of would be to always keep a version in memory and that's almost the same as a long ttl.
If you've got a better idea please raise a feature request ticket.

**Comment by khaled4vokalz** [comment]
It seems like it is possible to server STALE content from the plugin, adding background_update support requires a feature request I guess. I'll see if I get the time to propose something as a feature request. Closing this discussion for now.
